agent/nrpe.rb

```diff
@@ -46,17 +46,25 @@ class Nrpe<RPC::Agent
       # If the command does not exist run will return exitcode 3.
       #
       # The Nrpe configuration directory and file containing checks
-      # must be specified in server.cfg
+      # must be specified in server.cfg. In here, the user who should run
+      # the check can also be specified.
       #
       # Example :
       #          plugin.nrpe.conf_dir = /etc/nagios/nrpe
       #          plugin.nrpe.conf_file = checks.nrpe
+      #          plugin.nrpe.runas_user = nrpe
       def self.run(command, args=[])
         nrpe_command = Nrpe.plugin_for_command(command, args)
 
         return 3, "No such command: #{command}" unless nrpe_command
 
         output = ""
+        config = Config.instance
+        runas_user = config.pluginconf["nrpe.runas_user"] || nil
```

Comment:
`runas_user = @config.pluginconf["nrpe.runas_user"]`, you dont need to set config and dont need the `|| nil` :)

Reply:
@config is not available

Reply:
ah doh, its because this is a class method not an instance one - you're right, nevermind me :)